### PR TITLE
[t1420] scrolling fixes in timeline module

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5264c3ba9cac372/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/aa6ec2bdf2a7cb2/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5264c3ba9cac372/output.txt

Total script time: 0.28 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/aa6ec2bdf2a7cb2/output.txt

Total script time: 0.52 mins

Server is running on http://pr453.butyr.org/templates/test.html

File: src/timeline/track-container.js [hunk 1]
```diff
@@ -49,6 +52,11 @@ function(
       }
     });
 
+    this.setScrollbars = function ( hScrollbar, vScrollbar ) {
+      _hScrollbar = hScrollbar;
+      _vScrollbar = vScrollbar;
+    };
+
     this.orderTracks = function( orderedTracks ){
```
Comment (ScottDowne) [inline src/timeline/track-container.js:60]: This line:

this.orderTracks = function( orderedTracks ){

Has differnet whitespace than this line:

this.setScrollbars = function ( hScrollbar, vScrollbar ) {

File: src/timeline/track-container.js [hunk 2]
```diff
@@ -109,6 +118,9 @@ function(
     _media.listen( "trackremoved", function( e ){
       var trackView = e.data.view;
       _container.removeChild( trackView.element );
+      if ( _vScrollbar ) {
+        _vScrollbar.update();
+      }
```
Comment (ScottDowne) [inline src/timeline/track-container.js:123]: See above comment.

File: src/timeline/track-container.js [hunk 3]
```diff
@@ -21,6 +21,9 @@ function(
     var _element = document.createElement( "div" ),
         _container = document.createElement( "div" );
 
+    var _hScrollbar,
+        _vScrollbar;
+
     var _droppable;
```
Comment (ScottDowne) [inline src/timeline/track-container.js:27]: Should do:

var _hScrollbar,
       _vScrollbar,
       _droppable;

File: src/timeline/scrollbars.js [hunk 4]
```diff
@@ -68,10 +64,11 @@ define( [ "core/eventmanager" ], function( EventManagerWrapper ){
...
-      if( _containerChild.scrollHeight - _elementHeight > 0 ) {
+      if ( _containerChild.scrollHeight - _elementHeight > 0 ) {
         _handle.style.top = ( _elementHeight - _handleHeight ) *
-          ( _containerParent.scrollTop / ( _containerChild.scrollHeight - _elementHeight ) ) + "px";
-      }else{
+          ( _containerParent.scrollTop / ( _containerChild.scrollHeight - _parentHeight ) ) + "px";
+      }
+      else {
         _handle.style.top = "0px";
       }
```
Comment (ScottDowne) [inline src/timeline/scrollbars.js:73]: See second down comment on whitespace issues.

File: src/timeline/track-container.js (hunk 3 above)
Comment (secretrobotron) [inline src/timeline/track-container.js:27]: I kept these separated on purpose just to group them.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d8d3c04826e7d38/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/32929d2d34d4ec4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d8d3c04826e7d38/output.txt

Total script time: 0.45 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/32929d2d34d4ec4/output.txt

Total script time: 0.88 mins

Server is running on http://pr453.butyr.org/templates/test.html

**Pull request merged**